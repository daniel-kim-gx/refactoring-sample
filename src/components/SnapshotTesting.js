import Bolder from "../base/Bolder";

function SnapshotTesting({ list, object }) {
  return (
    <div>
      THIS component is for snapshot testing.
      <Child1 list={list} />
      <Child2 object={object} />
    </div>
  );
}

function Child1({ list }) {
  return (
    <div>
      This is child one.
      <ul>
        {list.map((item) => {
          return (
            <li key={item.id}>
              <Bolder>Title is : {item.title}</Bolder>
            </li>
          );
        })}
      </ul>
      <LengthChecker list={list} />
    </div>
  );
}

function Child2({ object }) {
  const { name, age, address } = object;

  return (
    <div>
      This is child two.
      <div>
        <Bolder>
          My name is: <strong>{name}</strong>
        </Bolder>
      </div>
      <div>
        My age is: <strong>{age}</strong>
      </div>
      <div>
        My address is: <strong>{address}</strong>
      </div>
    </div>
  );
}

function LengthChecker({ list }) {
  return <div>List's length is : {list.length}</div>;
}

export default SnapshotTesting;
