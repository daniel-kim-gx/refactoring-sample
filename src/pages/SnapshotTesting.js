import SnapshotTestingComponent from "../components/SnapshotTesting";

const list = [
  {
    id: "1",
    title: "This is title 1",
    description: "This is description 1",
  },

  {
    id: "2",
    title: "This is title 2",
    description: "This is description 2",
  },

  {
    id: "3",
    title: "This is title 3",
    description: "This is description 3",
  },
];

const object = {
  name: "This is real name",
  age: 400,
  Address: "This is real address",
};

const SnapshotTesting = () => {
  return <SnapshotTestingComponent list={list} object={object} />;
};

export default SnapshotTesting;
