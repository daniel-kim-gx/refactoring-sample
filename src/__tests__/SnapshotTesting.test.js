import { render } from "@testing-library/react";
import SnapshotTesting from "../components/SnapshotTesting";

const mockProps = {
  list: [
    {
      id: "test-id-1",
      title: "test-title-1",
      description: "test-description-1",
    },

    {
      id: "test-id-2",
      title: "test-title-2",
      description: "test-description-2",
    },

    {
      id: "test-id-3",
      title: "test-title-3",
      description: "test-description-3",
    },
  ],

  object: {
    name: "test-name",
    age: 400,
    address: "test-address",
  },
};

describe("<SnapshotTesting />", () => {
  it("Should match snapshot", () => {
    const { container } = render(<SnapshotTesting {...mockProps} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
