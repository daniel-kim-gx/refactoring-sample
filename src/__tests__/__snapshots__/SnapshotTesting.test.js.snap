// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SnapshotTesting /> Should match snapshot 1`] = `
<div>
  THIS component is for snapshot testing.
  <div>
    This is child one.
    <ul>
      <li>
        <b>
          Title is : 
          test-title-1
        </b>
      </li>
      <li>
        <b>
          Title is : 
          test-title-2
        </b>
      </li>
      <li>
        <b>
          Title is : 
          test-title-3
        </b>
      </li>
    </ul>
    <div>
      List's length is : 
      3
    </div>
  </div>
  <div>
    This is child two.
    <div>
      <b>
        My name is: 
        <strong>
          test-name
        </strong>
      </b>
    </div>
    <div>
      My age is: 
      <strong>
        400
      </strong>
    </div>
    <div>
      My address is: 
      <strong>
        test-address
      </strong>
    </div>
  </div>
</div>
`;
